fcomm/src/lib.rs

```diff
@@ -1085,6 +1086,48 @@ mod test {
     use lurk::public_parameters::public_params;
     use lurk::z_data::{from_z_data, to_z_data};
 
+    // You have broken a snapshot test. Unlike round-trip tests, those tests check the actual format of serialized Lurk expressions,
+    // and since you broke one, it's probable that you have changed that format, which will break at least the fcomm examples.
+    // Please read the documentation on snapshot tests `https://insta.rs/docs/quickstart/`, fix the snapshot **AND**
+    // make sure `cargo nextest run --run-ignored all -E 'test(test_make_fcomm_examples)'` passes
+    #[test]
+    fn test_snapshot_serialized_expressions() {
+        let function_source: &str = "(letrec ((secret 12345) (a (lambda (acc x) (let ((acc (+ acc x))) (cons acc (hide secret (a acc))))))) (a 0))";
+        let function_inputs: &str = "(+ 1 2)";
+        let committed_expression = CommittedExpression::<S1> {
+            expr: LurkPtr::Source(function_source.into()),
+            secret: None,
+            commitment: None,
+        };
+        assert_json_snapshot!(committed_expression);
+
+        let input = Expression::<S1> {
+            expr: LurkPtr::Source(function_inputs.into()),
+        };
+        assert_json_snapshot!(input);
+
+        let c = Commitment {
+            comm: S1::from(123),
+        };
+        assert_json_snapshot!(c);
```

Comment:
It's quite confusing to manually create a commitment with a value that doesn't correspond to a commitment.

Reply:
I think snapshot testing would be better served by creating real commitments. Then it would be observed to break when hashing or some other element of the commitment-production chain actually changes — which is one of the most important things we should be tracking (and eventually preventing at all costs).

Reply:
> I think snapshot testing would be better served by creating real commitments. Then it would be observed to break when hashing or some other element of the commitment-production chain actually changes — which is one of the most important things we should be tracking (and eventually preventing at all costs).

This is the purpose of the [fcomm tests](https://github.com/lurk-lab/lurk-rs/blob/master/fcomm/tests/makefile_tests.rs). Those are run on [every push](https://github.com/lurk-lab/lurk-rs/blob/master/.github/workflows/rust.yml#L55).